builds.js

```diff
@@ -9,112 +12,327 @@ const environments = new Map([
   ['dist', 'prod'],
   ['prod', 'prod']
 ]);
+const defaultCacheRefreshInterval = 20 * 60 * 60 * 1000;
+const defaultCacheRefreshLimit = 10;
 
 /**
  * Build interface for the Warehouse API.
  *
- * @constructor
- * @param {Warehouse} warehouse Reference to the Warehouse instance.
- * @api public
+ * @class Builds
  */
-function Builds(warehouse) {
-  this.warehouse = warehouse;
-}
+class Builds {
+  /**
+   * @constructor
+   * @param {Warehouse} warehouse Reference to the Warehouse instance.
+   * @param {Object} [options] Warehouse options object
+   * @param {Object} [options.buildCache] An options object for caching build information
+   * @param {boolean} [options.buildCache.enabled] True if you want to use a cache of build information,
+   * False if you want to hit the service every time
+   * @param {number} [options.buildCache.refreshLimit] The number of cache entries that can be refreshed simultaneously.
+   * @param {number} [options.buildCache.refreshInterval] An interval in ms on which the cache should be refreshed.
+   * If the refresh takes longer than the interval, intervals will be skipped. A non-positive number results in the
+   * cache _never_ being refreshed
+   * @public
+   */
+  constructor(warehouse, options) {
```

Comment:
so in the main `warehouse.ai-api-client` constructor, lets make a separate `build` key on those options and lets have that have a `cache` key. So the options passed into this constructor are `options.build` since any options we need from the `warehouse` instance would be available to us on the instance below. This now enables us to deal with `options.cache` in the context of this `Builds` class.